### ISSUE TITLE
 Logical error: 'current_thread is not initialized'.

### ISSUE THREAD

**Issue by qoega** [ClickHouse/ClickHouse#52013]
https://s3.amazonaws.com/clickhouse-test-reports/0/344bc05156c6336a98bfbacc613ad8675939587d/stress_test__tsan_.html
```
 Logical error: 'current_thread is not initialized'.
BaseDaemon: ########## Short fault info ############
BaseDaemon: (version 23.7.1.1 (official build), build id: 2D4D30D16F4DF24B51A65AE5416FC15DADF6777E, git hash: 344bc05156c6336a98bfbacc613ad8675939587d) (from thread 4437) Received signal 6
BaseDaemon: Signal description: Aborted
BaseDaemon: 
BaseDaemon: Stack trace: 0x00007f25a735ea7c 0x00007f25a730a476 0x00007f25a72f07f3 0x000055bb1cc925a3 0x000055bb25063369 0x000055bb25063c40 0x000055bb1d31bbc6 0x000055bb25334355 0x000055bb253356b2 0x000055bb25331277 0x000055bb2532f720 0x000055bb2ce8d6b3 0x000055bb2ce8c486 0x000055bb2ce8c217 0x000055bb3425e1f8 0x000055bb3425b01f 0x000055bb342790d3 0x000055bb34279048 0x000055bb3434a972 0x000055bb2ce6bace 0x000055bb2da0e672 0x000055bb2da0f066 0x000055bb2da41956 0x000055bb2da3c44c 0x000055bb2da1659a 0x000055bb2fa15702 0x000055bb2fa11c33 0x000055bb2fa2ee00 0x000055bb2fafecfd 0x000055bb2fafc1f1 0x000055bb2fafba27 0x000055bb2fb081df 0x000055bb2fee530e 0x000055bb2d91fe86 0x000055bb2dc97e90 0x000055bb2dc2027e 0x000055bb2dc14b56 0x000055bb2dbd4d94 0x000055bb2dc1db4d 0x000055bb2ea54110
BaseDaemon: ########################################
BaseDaemon: (version 23.7.1.1 (official build), build id: 2D4D30D16F4DF24B51A65AE5416FC15DADF6777E, git hash: 344bc05156c6336a98bfbacc613ad8675939587d) (from thread 4437) (no query) Received signal Aborted (6)
BaseDaemon: 
BaseDaemon: Stack trace: 0x00007f25a735ea7c 0x00007f25a730a476 0x00007f25a72f07f3 0x000055bb1cc925a3 0x000055bb25063369 0x000055bb25063c40 0x000055bb1d31bbc6 0x000055bb25334355 0x000055bb253356b2 0x000055bb25331277 0x000055bb2532f720 0x000055bb2ce8d6b3 0x000055bb2ce8c486 0x000055bb2ce8c217 0x000055bb3425e1f8 0x000055bb3425b01f 0x000055bb342790d3 0x000055bb34279048 0x000055bb3434a972 0x000055bb2ce6bace 0x000055bb2da0e672 0x000055bb2da0f066 0x000055bb2da41956 0x000055bb2da3c44c 0x000055bb2da1659a 0x000055bb2fa15702 0x000055bb2fa11c33 0x000055bb2fa2ee00 0x000055bb2fafecfd 0x000055bb2fafc1f1 0x000055bb2fafba27 0x000055bb2fb081df 0x000055bb2fee530e 0x000055bb2d91fe86 0x000055bb2dc97e90 0x000055bb2dc2027e 0x000055bb2dc14b56 0x000055bb2dbd4d94 0x000055bb2dc1db4d 0x000055bb2ea54110
BaseDaemon: 5. pthread_kill @ 0x00007f25a735ea7c in ?
BaseDaemon: 6. gsignal @ 0x00007f25a730a476 in ?
BaseDaemon: 7. abort @ 0x00007f25a72f07f3 in ?
BaseDaemon: 8. __interceptor_abort @ 0x000000000dcf95a3 in /usr/bin/clickhouse
BaseDaemon: 9. ./build_docker/./src/Common/Exception.cpp:0: DB::abortOnFailedAssertion(String const&) @ 0x00000000160ca369 in /usr/bin/clickhouse
BaseDaemon: 10.1. inlined from ./build_docker/./src/Common/Exception.cpp:66: DB::handle_error_code(String const&, int, bool, std::vector<void*, std::allocator<void*>> const&)
BaseDaemon: 10. ./build_docker/./src/Common/Exception.cpp:96: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x00000000160cac40 in /usr/bin/clickhouse
BaseDaemon: 11. DB::Exception::Exception<char const (&) [34], void>(int, char const (&) [34]) @ 0x000000000e382bc6 in /usr/bin/clickhouse
BaseDaemon: 12.1. inlined from ./build_docker/./src/Common/MemoryTrackerSwitcher.h:19: MemoryTrackerSwitcher
BaseDaemon: 12. ./build_docker/./src/IO/HTTPCommon.cpp:114: DB::(anonymous namespace)::SingleEndpointHTTPSessionPool::allocObject() @ 0x000000001639b355 in /usr/bin/clickhouse
BaseDaemon: 13. ./build_docker/./src/Common/PoolBase.h:180: PoolBase<Poco::Net::HTTPClientSession>::get(long) @ 0x000000001639c6b2 in /usr/bin/clickhouse
BaseDaemon: 14.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:797: std::__variant_detail::__base<(std::__variant_detail::_Trait)1, std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::index[abi:v15000]() const
BaseDaemon: 14.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:792: std::__variant_detail::__base<(std::__variant_detail::_Trait)1, std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::valueless_by_exception[abi:v15000]() const
BaseDaemon: 14.3. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1446: std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::valueless_by_exception[abi:v15000]() const
BaseDaemon: 14.4. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1699: void std::__throw_if_valueless[abi:v15000]<std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>> const&>(std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>> const&)
BaseDaemon: 14.5. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1712: decltype(auto) std::visit[abi:v15000]<PoolBase<Poco::Net::HTTPClientSession>::Entry::castToObjectPtr() const::'lambda'(auto const&), std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>> const&, void>(auto&&, std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>> const&)
BaseDaemon: 14.6. inlined from ./build_docker/./src/Common/PoolBase.h:130: PoolBase<Poco::Net::HTTPClientSession>::Entry::castToObjectPtr() const
BaseDaemon: 14.7. inlined from ./build_docker/./src/Common/PoolBase.h:102: PoolBase<Poco::Net::HTTPClientSession>::Entry::operator->() &
BaseDaemon: 14.8. inlined from ./build_docker/./src/IO/HTTPCommon.cpp:276: DB::(anonymous namespace)::HTTPSessionPool::getSession(Poco::URI const&, Poco::URI const&, DB::ConnectionTimeouts const&, unsigned long, bool, bool)
BaseDaemon: 14. ./build_docker/./src/IO/HTTPCommon.cpp:334: DB::makePooledHTTPSession(Poco::URI const&, Poco::URI const&, DB::ConnectionTimeouts const&, unsigned long, bool, bool) @ 0x0000000016398277 in /usr/bin/clickhouse
BaseDaemon: 15. ./build_docker/./src/IO/HTTPCommon.cpp:0: DB::makePooledHTTPSession(Poco::URI const&, DB::ConnectionTimeouts const&, unsigned long, bool, bool) @ 0x0000000016396720 in /usr/bin/clickhouse
BaseDaemon: 16.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:797: std::__variant_detail::__base<(std::__variant_detail::_Trait)1, std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::index[abi:v15000]() const
BaseDaemon: 16.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:792: std::__variant_detail::__base<(std::__variant_detail::_Trait)1, std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::valueless_by_exception[abi:v15000]() const
BaseDaemon: 16.3. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1033: void std::__variant_detail::__assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>>::__generic_assign[abi:v15000]<std::__variant_detail::__move_assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>, (std::__variant_detail::_Trait)1>>(std::__variant_detail::__move_assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>, (std::__variant_detail::_Trait)1>&&)
BaseDaemon: 16.4. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1076: std::__variant_detail::__move_assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>, (std::__variant_detail::_Trait)1>::operator=(std::__variant_detail::__move_assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>, (std::__variant_detail::_Trait)1>&&)
BaseDaemon: 16.5. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1117: std::__variant_detail::__copy_assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>, (std::__variant_detail::_Trait)1>::operator=(std::__variant_detail::__copy_assignment<std::__variant_detail::__traits<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>, (std::__variant_detail::_Trait)1>&&)
BaseDaemon: 16.6. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1140: std::__variant_detail::__impl<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::operator=(std::__variant_detail::__impl<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>&&)
BaseDaemon: 16.7. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/variant:1378: std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>::operator=(std::variant<std::shared_ptr<Poco::Net::HTTPClientSession>, std::shared_ptr<PoolBase<Poco::Net::HTTPClientSession>::PoolEntryHelper>>&&)
BaseDaemon: 16.8. inlined from ./build_docker/./src/Common/PoolBase.h:84: PoolBase<Poco::Net::HTTPClientSession>::Entry::operator=(PoolBase<Poco::Net::HTTPClientSession>::Entry&&)
BaseDaemon: 16. ./build_docker/./src/IO/S3/PocoHTTPClient.cpp:354: void DB::S3::PocoHTTPClient::makeRequestInternalImpl<true>(Aws::Http::HttpRequest&, DB::S3::ClientConfigurationPerRequest const&, std::shared_ptr<DB::S3::PocoHTTPResponse>&, Aws::Utils::RateLimits::RateLimiterInterface*, Aws::Utils::RateLimits::RateLimiterInterface*) const @ 0x000000001def46b3 in /usr/bin/clickhouse
BaseDaemon: 17.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: String::__is_long[abi:v15000]() const
BaseDaemon: 17.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:2333: ~basic_string
BaseDaemon: 17.3. inlined from ./build_docker/./src/IO/S3/PocoHTTPClient.h:37: ~ClientConfigurationPerRequest
BaseDaemon: 17. ./build_docker/./src/IO/S3/PocoHTTPClient.cpp:270: DB::S3::PocoHTTPClient::makeRequestInternal(Aws::Http::HttpRequest&, std::shared_ptr<DB::S3::PocoHTTPResponse>&, Aws::Utils::RateLimits::RateLimiterInterface*, Aws::Utils::RateLimits::RateLimiterInterface*) const @ 0x000000001def3486 in /usr/bin/clickhouse
BaseDaemon: 18. ./build_docker/./src/IO/S3/PocoHTTPClient.cpp:0: DB::S3::PocoHTTPClient::MakeRequest(std::shared_ptr<Aws::Http::HttpRequest> const&, Aws::Utils::RateLimits::RateLimiterInterface*, Aws::Utils::RateLimits::RateLimiterInterface*) const @ 0x000000001def3217 in /usr/bin/clickhouse
BaseDaemon: 19. Aws::Client::AWSClient::AttemptOneRequest(std::shared_ptr<Aws::Http::HttpRequest> const&, Aws::AmazonWebServiceRequest const&, char const*, char const*, char const*) const @ 0x00000000252c51f8 in /usr/bin/clickhouse
BaseDaemon: 20. Aws::Client::AWSClient::AttemptExhaustively(Aws::Http::URI const&, Aws::AmazonWebServiceRequest const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const @ 0x00000000252c201f in /usr/bin/clickhouse
BaseDaemon: 21. Aws::Client::AWSXMLClient::MakeRequest(Aws::Http::URI const&, Aws::AmazonWebServiceRequest const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const @ 0x00000000252e00d3 in /usr/bin/clickhouse
BaseDaemon: 22. Aws::Client::AWSXMLClient::MakeRequest(Aws::AmazonWebServiceRequest const&, Aws::Endpoint::AWSEndpoint const&, Aws::Http::HttpMethod, char const*, char const*, char const*) const @ 0x00000000252e0048 in /usr/bin/clickhouse
BaseDaemon: 23. Aws::S3::S3Client::DeleteObjects(Aws::S3::Model::DeleteObjectsRequest const&) const @ 0x00000000253b1972 in /usr/bin/clickhouse
BaseDaemon: 24.1. inlined from ./build_docker/./src/IO/S3/Client.cpp:0: std::invoke_result<DB::S3::Client::DeleteObjects(DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest> const&) const::$_0, DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest>>::type DB::S3::Client::doRequest<DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest>, DB::S3::Client::DeleteObjects(DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest> const&) const::$_0>(DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest> const&, DB::S3::Client::DeleteObjects(DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest> const&) const::$_0) const
BaseDaemon: 24. ./build_docker/./src/IO/S3/Client.cpp:431: DB::S3::Client::DeleteObjects(DB::S3::ExtendedRequest<Aws::S3::Model::DeleteObjectsRequest> const&) const @ 0x000000001ded2ace in /usr/bin/clickhouse
BaseDaemon: 25.1. inlined from ./build_docker/./contrib/aws/src/aws-cpp-sdk-core/include/aws/core/utils/Outcome.h:160: Aws::Utils::Outcome<Aws::S3::Model::DeleteObjectsResult, Aws::S3::S3Error>::IsSuccess() const
BaseDaemon: 25.2. inlined from ./build_docker/./src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:71: void DB::(anonymous namespace)::throwIfUnexpectedError<Aws::S3::Model::DeleteObjectsResult, Aws::S3::S3Error>(Aws::Utils::Outcome<Aws::S3::Model::DeleteObjectsResult, Aws::S3::S3Error> const&, bool)
BaseDaemon: 25. ./build_docker/./src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:369: DB::S3ObjectStorage::removeObjectsImpl(std::vector<DB::StoredObject, std::allocator<DB::StoredObject>> const&, bool) @ 0x000000001ea75672 in /usr/bin/clickhouse
BaseDaemon: 26. ./build_docker/./src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp:394: DB::S3ObjectStorage::removeObjectsIfExist(std::vector<DB::StoredObject, std::allocator<DB::StoredObject>> const&) @ 0x000000001ea76066 in /usr/bin/clickhouse
BaseDaemon: 27. ./build_docker/./src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp:227: DB::(anonymous namespace)::RemoveManyObjectStorageOperation::finalize() @ 0x000000001eaa8956 in /usr/bin/clickhouse
BaseDaemon: 28.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__iterator/wrap_iter.h:100: std::__wrap_iter<std::unique_ptr<DB::IDiskObjectStorageOperation, std::default_delete<DB::IDiskObjectStorageOperation>>*>::operator++[abi:v15000]()
BaseDaemon: 28. ./build_docker/./src/Disks/ObjectStorages/DiskObjectStorageTransaction.cpp:812: DB::DiskObjectStorageTransaction::commit() @ 0x000000001eaa344c in /usr/bin/clickhouse
BaseDaemon: 29.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:701: ~shared_ptr
BaseDaemon: 29. ./build_docker/./src/Disks/ObjectStorages/DiskObjectStorage.cpp:224: DB::DiskObjectStorage::removeSharedFiles(std::vector<DB::RemoveRequest, std::allocator<DB::RemoveRequest>> const&, bool, std::unordered_set<String, std::hash<String>, std::equal_to<String>, std::allocator<String>> const&) @ 0x000000001ea7d59a in /usr/bin/clickhouse
BaseDaemon: 30. ./build_docker/./src/Storages/MergeTree/DataPartStorageOnDiskBase.cpp:0: DB::DataPartStorageOnDiskBase::clearDirectory(String const&, DB::CanRemoveDescription const&, DB::MergeTreeDataPartChecksums const&, bool, Poco::Logger*) @ 0x0000000020a7c702 in /usr/bin/clickhouse
BaseDaemon: 31.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: String::__is_long[abi:v15000]() const
BaseDaemon: 31.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:2333: ~basic_string
BaseDaemon: 31. ./build_docker/./src/Storages/MergeTree/DataPartStorageOnDiskBase.cpp:701: DB::DataPartStorageOnDiskBase::remove(std::function<DB::CanRemoveDescription ()>&&, DB::MergeTreeDataPartChecksums const&, std::list<DB::IDataPartStorage::ProjectionChecksums, std::allocator<DB::IDataPartStorage::ProjectionChecksums>>, bool, Poco::Logger*) @ 0x0000000020a78c33 in /usr/bin/clickhouse
BaseDaemon: 32.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/list:616: std::__list_imp<DB::IDataPartStorage::ProjectionChecksums, std::allocator<DB::IDataPartStorage::ProjectionChecksums>>::empty[abi:v15000]() const
BaseDaemon: 32.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/list:732: std::__list_imp<DB::IDataPartStorage::ProjectionChecksums, std::allocator<DB::IDataPartStorage::ProjectionChecksums>>::clear()
BaseDaemon: 32.3. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/list:724: ~__list_imp
BaseDaemon: 32. ./build_docker/./src/Storages/MergeTree/IMergeTreeDataPart.cpp:1742: DB::IMergeTreeDataPart::remove() @ 0x0000000020a95e00 in /usr/bin/clickhouse
BaseDaemon: 33.1. inlined from ./build_docker/./src/Storages/MergeTree/MergeTreeData.cpp:2381: ~shared_ptr
BaseDaemon: 33. ./build_docker/./src/Storages/MergeTree/MergeTreeData.cpp:2381: DB::MergeTreeData::clearPartsFromFilesystemImpl(std::vector<std::shared_ptr<DB::IMergeTreeDataPart const>, std::allocator<std::shared_ptr<DB::IMergeTreeDataPart const>>> const&, std::unordered_set<String, std::hash<String>, std::equal_to<String>, std::allocator<String>>*)::$_3::operator()() const @ 0x0000000020b65cfd in /usr/bin/clickhouse
BaseDaemon: 34.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:701: ~shared_ptr
BaseDaemon: 34. ./build_docker/./src/Storages/MergeTree/MergeTreeData.cpp:2602: DB::MergeTreeData::clearPartsFromFilesystemImpl(std::vector<std::shared_ptr<DB::IMergeTreeDataPart const>, std::allocator<std::shared_ptr<DB::IMergeTreeDataPart const>>> const&, std::unordered_set<String, std::hash<String>, std::equal_to<String>, std::allocator<String>>*) @ 0x0000000020b631f1 in /usr/bin/clickhouse
BaseDaemon: 35.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__hash_table:935: std::__hash_table<String, std::hash<String>, std::equal_to<String>, std::allocator<String>>::size[abi:v15000]() const
BaseDaemon: 35.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/unordered_set:638: std::unordered_set<String, std::hash<String>, std::equal_to<String>, std::allocator<String>>::size[abi:v15000]() const
BaseDaemon: 35.3. inlined from ./build_docker/./src/Storages/MergeTree/MergeTreeData.cpp:2339: operator()
BaseDaemon: 35. ./build_docker/./src/Storages/MergeTree/MergeTreeData.cpp:2355: DB::MergeTreeData::clearPartsFromFilesystem(std::vector<std::shared_ptr<DB::IMergeTreeDataPart const>, std::allocator<std::shared_ptr<DB::IMergeTreeDataPart const>>> const&, bool, std::unordered_set<String, std::hash<String>, std::equal_to<String>, std::allocator<String>>*) @ 0x0000000020b62a27 in /usr/bin/clickhouse
BaseDaemon: 36. ./build_docker/./src/Storages/MergeTree/MergeTreeData.cpp:0: DB::MergeTreeData::dropAllData() @ 0x0000000020b6f1df in /usr/bin/clickhouse
BaseDaemon: 37. ./build_docker/./src/Storages/StorageMergeTree.cpp:305: DB::StorageMergeTree::drop() @ 0x0000000020f4c30e in /usr/bin/clickhouse
BaseDaemon: 38. ./build_docker/./src/Databases/DatabaseMemory.cpp:77: DB::DatabaseMemory::dropTable(std::shared_ptr<DB::Context const>, String const&, bool) @ 0x000000001e986e86 in /usr/bin/clickhouse
BaseDaemon: 39.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: String::__is_long[abi:v15000]() const
BaseDaemon: 39.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:2333: ~basic_string
BaseDaemon: 39. ./build_docker/./src/Interpreters/DatabaseCatalog.cpp:134: DB::TemporaryTableHolder::~TemporaryTableHolder() @ 0x000000001ecfee90 in /usr/bin/clickhouse
BaseDaemon: 40. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:313: std::__shared_ptr_emplace<DB::TemporaryTableHolder, std::allocator<DB::TemporaryTableHolder>>::__on_zero_shared() @ 0x000000001ec8727e in /usr/bin/clickhouse
BaseDaemon: 41.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:216: std::__shared_weak_count::__release_shared[abi:v15000]()
BaseDaemon: 41.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:702: ~shared_ptr
BaseDaemon: 41.3. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__utility/pair.h:40: ~pair
BaseDaemon: 41.4. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/construct_at.h:63: void std::__destroy_at[abi:v15000]<std::pair<String const, std::shared_ptr<DB::TemporaryTableHolder>>, 0>(std::pair<String const, std::shared_ptr<DB::TemporaryTableHolder>>*)
BaseDaemon: 41.5. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/construct_at.h:88: void std::destroy_at[abi:v15000]<std::pair<String const, std::shared_ptr<DB::TemporaryTableHolder>>, 0>(std::pair<String const, std::shared_ptr<DB::TemporaryTableHolder>>*)
BaseDaemon: 41.6. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/allocator_traits.h:317: void std::allocator_traits<std::allocator<std::__tree_node<std::__value_type<String, std::shared_ptr<DB::TemporaryTableHolder>>, void*>>>::destroy[abi:v15000]<std::pair<String const, std::shared_ptr<DB::TemporaryTableHolder>>, void, void>(std::allocator<std::__tree_node<std::__value_type<String, std::shared_ptr<DB::TemporaryTableHolder>>, void*>>&, std::pair<String const, std::shared_ptr<DB::TemporaryTableHolder>>*)
BaseDaemon: 41. ./build_docker/./contrib/llvm-project/libcxx/include/__tree:1800: std::__tree<std::__value_type<String, std::shared_ptr<DB::TemporaryTableHolder>>, std::__map_value_compare<String, std::__value_type<String, std::shared_ptr<DB::TemporaryTableHolder>>, std::less<String>, true>, std::allocator<std::__value_type<String, std::shared_ptr<DB::TemporaryTableHolder>>>>::destroy(std::__tree_node<std::__value_type<String, std::shared_ptr<DB::TemporaryTableHolder>>, void*>*) @ 0x000000001ec7bb56 in /usr/bin/clickhouse
BaseDaemon: 42.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: String::__is_long[abi:v15000]() const
BaseDaemon: 42.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/string:2333: ~basic_string
BaseDaemon: 42. ./build_docker/./src/Interpreters/Context.cpp:718: DB::Context::~Context() @ 0x000000001ec3bd94 in /usr/bin/clickhouse
BaseDaemon: 43.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/unique_ptr.h:48: std::default_delete<DB::Context>::operator()[abi:v15000](DB::Context*) const
BaseDaemon: 43. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:263: std::__shared_ptr_pointer<DB::Context*, std::shared_ptr<DB::Context>::__shared_ptr_default_delete<DB::Context, DB::Context>, std::allocator<DB::Context>>::__on_zero_shared() @ 0x000000001ec84b4d in /usr/bin/clickhouse
BaseDaemon: 44.1. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:216: std::__shared_weak_count::__release_shared[abi:v15000]()
BaseDaemon: 44.2. inlined from ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:702: ~shared_ptr
BaseDaemon: 44. ./build_docker/./src/Interpreters/Session.cpp:304: DB::Session::~Session() @ 0x000000001fabb110 in /usr/bin/clickhouse
BaseDaemon: Integrity check of the executable successfully passed (checksum: BB9DB2FDB66BDC8CD1234F387B270021)
BaseDaemon: Report this error to https://github.com/ClickHouse/ClickHouse/issues
Application: Child process was terminated by signal 6.
```

**Comment by qoega** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/4666838a5d97f9108c787bf4c4ea3e9c11f160ad/stress_test__ubsan_.html

**Comment by qoega** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/2492f1bb783979c49ffd2e3741f8f4760b3d65f1/stress_test__tsan_.html

**Comment by qoega** [comment]
https://s3.amazonaws.com/clickhouse-test-reports/0/49c1beb8705399dfb40ae5213ebca70ba31e3852/stress_test__ubsan_.html

**Comment by azat** [comment]
This has been fixed in [`3c18a18` (#51448)](https://github.com/ClickHouse/ClickHouse/pull/51448/commits/3c18a181c997f1f43e759d72eeadcc5d4f35142d)

**Comment by azat** [comment]
https://github.com/ClickHouse/ClickHouse/pull/49732#discussion_r1257469094

**Comment by azat** [comment]
>This has been fixed in [3c18a18 (#51448)](https://github.com/ClickHouse/ClickHouse/pull/51448/commits/3c18a181c997f1f43e759d72eeadcc5d4f35142d)

This should be fixed now. Closing.
